packages/@aws-cdk/cdk/lib/context.ts

```diff
@@ -34,6 +34,24 @@ export class Context {
     return new AvailabilityZoneProvider(scope).availabilityZones;
   }
 
+  /**
+   * Set the availability zone to be used in the given scope
+   * Given a scope it will set the AZs list in the first stack scope in which the given scope is defined
+   * Note that supplying an unresolved value such us the return value of getAZ() will throw
+   * @param scope Some construct scope.
+   * @param azs the list of availability zones
+   */
+  public static setAvailabilityZones(scope: Construct, azs: string[]) {
+    // Prevent using getAZ() return value which will result in an resolvable value
+    if (Token.isUnresolved(azs)) {
+      throw new Error(`The list of azs must hold concrate values, for using getAZs see:<link to example of using getAZs with select>`);
```

Comment:
It's not that the list of AZs must hold concrete values, it's that the array itself must be a concrete array. It's actually ok for the array to include unresolved values in individual elements.